core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -986,4 +990,26 @@ public boolean canAllow(CarbonTable carbonTable, TableOperation operation) {
     return true;
   }
 
+  /**
+   * Get all index columns specified by dataMapSchema
+   */
+  public List<CarbonColumn> getIndexedColumns(DataMapSchema dataMapSchema)
```

Comment:
I think it's better to use `columnNames` as the input parameter rather than `dataMapSchema`.
The method in CarbonTable class does not need to know more information than `columnNames` from `dataMapSchema`, so better to pass less information.

Reply:
Because there are 15 usages of this function, and they all has dataMapSchema. If I change to pass columnNames as input parameter, all 15 usages need to change and add more code, so I think it is better to keep in this way, like a helper function